### ISSUE TITLE
ClearAllCaches doesn't seem to be working in the same way since 1.0.9

### ISSUE THREAD

**Issue by fnalex** [imageworks/OpenColorIO#767]
Reloading a .ccc file doesn't seem to work the same in 1.0.9 and 1.1.x

The following code will create a file at /tmp/example.ccc and then create a CDL transform using the id cc03343. Then it will overwrite the file at /tmp/example.ccc with a different slope; clear all the caches and create a new transform from the same filepath with the same id.

```
#include <string>
#include <fstream>
#include <iostream>
#include <OpenColorIO/OpenColorIO.h>

namespace OCIO = OCIO_NAMESPACE;

static const std::string kContentsA = {
"<ColorCorrectionCollection>\n"
"    <ColorCorrection id=\"cc03343\">\n"
"        <SOPNode>\n"
"            <Slope>0.1 0.2 0.3 </Slope>\n"
"            <Offset>0.8 0.1 0.3 </Offset>\n"
"            <Power>0.5 0.5 0.5 </Power>\n"
"        </SOPNode>\n"
"        <SATNode>\n"
"            <Saturation>1</Saturation>\n"
"        </SATNode>\n"
"    </ColorCorrection>\n"
"    <ColorCorrection id=\"cc03344\">\n"
"        <SOPNode>\n"
"            <Slope>1.2 1.3 1.4 </Slope>\n"
"            <Offset>0.3 0 0 </Offset>\n"
"            <Power>0.75 0.75 0.75 </Power>\n"
"        </SOPNode>\n"
"        <SATNode>\n"
"            <Saturation>1</Saturation>\n"
"        </SATNode>\n"
"    </ColorCorrection>\n"
"</ColorCorrectionCollection>\n"
};

static const std::string kContentsB = {
"<ColorCorrectionCollection>\n"
"    <ColorCorrection id=\"cc03343\">\n"
"        <SOPNode>\n"
"            <Slope>1.1 2.2 3.3 </Slope>\n"
"            <Offset>0.8 0.1 0.3 </Offset>\n"
"            <Power>0.5 0.5 0.5 </Power>\n"
"        </SOPNode>\n"
"        <SATNode>\n"
"            <Saturation>1</Saturation>\n"
"        </SATNode>\n"
"    </ColorCorrection>\n"
"    <ColorCorrection id=\"cc03344\">\n"
"        <SOPNode>\n"
"            <Slope>1.2 1.3 1.4 </Slope>\n"
"            <Offset>0.3 0 0 </Offset>\n"
"            <Power>0.75 0.75 0.75 </Power>\n"
"        </SOPNode>\n"
"        <SATNode>\n"
"            <Saturation>1</Saturation>\n"
"        </SATNode>\n"
"    </ColorCorrection>\n"
"</ColorCorrectionCollection>\n"
};

int main()
{
  std::cout << "Using OCIO Version : " << OCIO_VERSION << std::endl;

  try {
    const auto filepath = "/tmp/example.ccc";

    std::fstream stream(filepath, std::ios_base::out);
    stream << kContentsA;
    stream.close();

    auto transform = OCIO::CDLTransform::CreateFromFile(filepath, "cc03343");

    std::array<float, 3> slope;
    transform->getSlope(slope.data());

    std::cout << "slopeA : " << slope[0] << ", " << slope[1] << ", " << slope[2] << "\n";

    stream.open(filepath, std::ios_base::out);
    stream << kContentsB;
    stream.close();

    OCIO::ClearAllCaches();
    transform = OCIO::CDLTransform::CreateFromFile(filepath, "cc03343");
    transform->getSlope(slope.data());

    std::cout << "slopeB : " << slope[0] << ", " << slope[1] << ", " << slope[2] << "\n";
  }
  catch (const OCIO::Exception& e) {
    std::cout << "[OCIO EXCEPTION] : " << e.what() << std::endl;
  }
  return 0;
}
```

I would expect the new transform to have the updated values but instead it throws an exception:

>Using OCIO Version : 1.0.9
>slopeA : 0.1, 0.2, 0.3
>slopeB : 1.1, 2.2, 3.3
>
>Using OCIO Version : 1.1.0
>slopeA : 0.1, 0.2, 0.3
>[OCIO EXCEPTION] : The specified cccid/cccindex 'cc03343' could not be loaded from the src file '/tmp/example.ccc'.
>
>Using OCIO Version : 1.1.1
>slopeA : 0.1, 0.2, 0.3
>[OCIO EXCEPTION] : The specified cccid/cccindex 'cc03343' could not be loaded from the src file '/tmp/example.ccc'.

Additionally it would be beneficial to avoid ::ClearAllCaches() and only clear (in this case) the transforms cache to avoid clearing all the valid caches.

**Comment by hodoulp** [comment]
> [OCIO EXCEPTION] : The specified cccid/cccindex 'cc03343' could not be loaded from the src file '/tmp/example.ccc'.

Concerning the exception only, I can reproduce it in `master`.

**Comment by hodoulp** [comment]
Merged in `master` i.e. OCIOv2

**Comment by hodoulp** [comment]
Merged in `RB-1.1` i.e. OCIOv1
